src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1650,22 +1663,49 @@ julia> flatten(df2, :b)
 │ 3   │ 2     │ r      │
 │ 4   │ 2     │ s      │
 
+julia> df3 = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]], c = [[5, 6], [7]])
+2×3 DataFrame
+│ Row │ a     │ b      │ c      │
+│     │ Int64 │ Array… │ Array… │
+├─────┼───────┼────────┼────────┤
+│ 1   │ 1     │ [1, 2] │ [5, 6] │
+│ 2   │ 2     │ [3, 4] │ [7]    │
+
+julia> flatten(df3, [:b, :c])
+ERROR: ArgumentError: Lengths of iterables stored in columns 2 and 3 are not the the same in
+row 2
 ```
 """
-function flatten(df::AbstractDataFrame, col::ColumnIndex)
-    col_to_flatten = df[!, col]
-    lengths = length.(col_to_flatten)
-    new_df = similar(df[!, Not(col)], sum(lengths))
+function flatten(df::AbstractDataFrame,
+                 cols::Union{ColumnIndex, AbstractVector, Regex, Not, Between, All, Colon})
+    _check_consistency(df)
+
+    idxcols = index(df)[cols]
+    isempty(idxcols) && return copy(df)
+    col1 = first(idxcols)
+    lengths = length.(df[!, col1])
+    for col in idxcols
+        v = df[!, col]
+        if any(x -> length(x[1]) != x[2], zip(v, lengths))
+            r = findfirst(x -> x != 0, length.(v) .- lengths)
+            throw(ArgumentError("Lengths of iterables stored in columns $col1 and $col are"
```

Comment:
It would be nice to print the name or indices of columns, as specified in `cols`, rather than always the indices.

Reply:
OK - now I see the ambiguity. However `cols` can be anything, so we should use column names then.